components/raftstore/src/store/entry_storage.rs

```diff
@@ -569,72 +569,67 @@ pub fn init_applied_term<ER: RaftEngine>(
 }
 
 /// When a peer(follower) receives a TransferLeaderMsg, it enters the
-/// CacheWarmupState. When the peer becomes leader or it doesn't
-/// become leader before a deadline, it exits the state.
+/// CacheWarmupState. And it exits the state if the peer becomes leader or
+/// the an election_timeout has passed.
+///
+/// There two distinct timeouts in the state:
+/// 1. Task timeout: the time to wait for fetching raft logs. After timeout, the
+///    peer responses a transfer leader regardless the task is finished.
+/// 2. Election timeout: the time to wait for preventing compact entry storage.
+///    This is required because changing the first index will invalid the warmed
+///    up raft log cache.
 #[derive(Clone, Debug)]
 pub struct CacheWarmupState {
     range: (u64, u64),
     is_task_timeout: bool,
-    is_stale: bool,
-    started_at: Instant,
+    task_timeout_at: Instant,
+    is_election_timeout: bool,
+    election_timeout_at: Instant,
```

Comment:
How about still keeping the start timestamp and check the timeout by `start_at.elapsed() > election_timeout` ?
```suggestion
    task_timeout_at: Duration,
    is_election_timeout: bool,
    election_timeout: Duration,
    start_at: Instant,
```

Reply:
I think they are almost identical while the current approach is simpler, and it save 8 bytes (the `start_at` field) and a method `elapsed()`.

Reply:
Reasonable. Reducing the number of extra fields will help to minimize memory usage and alleviate pressure on memory resources.